src/murmur/ServerDB.cpp

```diff
@@ -1981,22 +1981,50 @@ int Server::readLastChannel(int id) {
 	if (id < 0)
 		return -1;
 
+	if (!Meta::mp.bRememberChan)
+		return -1;
+
 	TransactionHolder th;
 	QSqlQuery &query = *th.qsqQuery;
 
-	SQLPREP("SELECT `lastchannel` FROM `%1users` WHERE `server_id` = ? AND `user_id` = ?");
+	SQLPREP("SELECT `lastchannel`,`last_seen` FROM `%1users` WHERE `server_id` = ? AND `user_id` = ?");
 	query.addBindValue(iServerNum);
 	query.addBindValue(id);
 	SQLEXEC();
 
 	if (query.next()) {
 		int cid = query.value(0).toInt();
-		if (qhChannels.contains(cid))
-			return cid;
+		QDateTime last_seen = QDateTime::fromString(query.value(1).toString(), Qt::ISODate);
+		last_seen.setTimeSpec(Qt::UTC);
+		QDateTime now = QDateTime::currentDateTime();
+		now = now.toTimeSpec(Qt::UTC);
+
+		int duration = Meta::mp.iRememberChanDuration;
+		if (duration <= 0 || last_seen.secsTo(now) <= duration) {
+			if (qhChannels.contains(cid))
+				return cid;
+		}
 	}
 	return -1;
 }
 
+void Server::setLastSeen(const User *p) {
+	if (p->iId < 0)
+		return;
```

Comment:
Please add a comment that we only want to store this information for registered users (otherwise it might be unclear, what this check is there for)